port/unix/omrsignal.c

```diff
@@ -1587,8 +1587,10 @@ initializeSignalTools(OMRPortLibrary *portLibrary)
 
 #if !defined(J9ZOS390)
 #if defined(OSX)
-	/* OSX only has named semaphores. They are not shared across processes, so unlink immediately. */
-	portLibrary->str_printf(portLibrary, semNames[0], 128, "/omr/wakeUpASyncReporter-%d", getpid());
+	/* OSX only has named semaphores. They are not shared across processes, so unlink immediately.
+	 * The semaphore name length must fit within the SEM_NAME_LEN (31) limit. 
+	 */
+	portLibrary->str_printf(portLibrary, semNames[0], 128, "/omr/wUASRptr%d-%u", getpid(), (uint16_t)portLibrary->time_nano_time(portLibrary));
```

Comment:
Instead of `128`, it should use `sizeof(semName)`.

If PIDs are 32-bits, this will be at most 30 characters, but if PIDs can be larger than 2^32, then it could be up to 39 characters in which case the prefix needs to shrink or some other strategy needs to be used to guarantee uniqueness.

If we use hex instead, the maximum lengths are smaller, but still potentially problematic if a PID needs a full 64 bits.

Reply:
sizeof(pid_t) is 4

Reply:
Any idea why using just the pid sometimes gives a "File exists" perror?

Reply:
Sorry, no theories about that.